execution_engine/src/core/runtime/mod.rs

```diff
@@ -1427,7 +1425,9 @@ where
         // Charge just for the amount that particular entry point cost - using gas cost from the
         // isolated runtime might have a recursive costs whenever system contract calls other system
         // contract.
-        self.gas(mint_runtime.gas_counter() - gas_counter)?;
+        let gas_diff = mint_runtime.gas_counter().used() - gas_counter.used();
+        debug_assert!(gas_diff.value() <= U512::from(u64::MAX));
+        self.gas(gas_diff.value().as_u64())?;
```

Comment:
Is this definitely always going to be safe to cast to a `u64`?  Before `self.gas()` was handling `U512` values.

Maybe we should look to remove the `debug_assert` and just do `self.context.charge_gas_large(gas_diff)?;` here?